_posts/2019-12-25-technical-debt.md

```diff
@@ -0,0 +1,107 @@
+
+From a practical standpoint, technical debt is fundamentally about coupling. Coupling prevents production of value from occuring indepdently, instead requiring careful coordination between producers. This coupling can occur at multiple areas in the software development process. It can manifest as a team being dependent on another team to perform some task before they perform their own task. It could involve needing to significantly modify a service before they can introduce a new feature. Reducing coupling increases production of business value, as it removes bottlenecks from the system. Whilst not reducing technical debt this does reduce it's impact. There are multiple ways software development organizations can rearrange themselves and their code in order to reduce coupling and contain the effects of technical debt.
+
+<br/>
+> Debt . . . . that peculiar nexus where money, narrative or story, and religious belief intersect, often with explosive force. - Margaret Atwood
+
+<br/>
+Technical debt is an infamous subject in software development. The common understanding is that we accrue it by some series of faustian pacts; short-cuts are taken in order to deliver a feature sooner, with the underlying assumption that additional features will take longer to complete. A development team that continues to accrue technical debt will eventually find time taken to deliver the next feature will begin to approach infinity. In defiance of what is an obvious problem to the average developer, business stakeholders are often loathe to address it. Technical debt is unyielding to measure and resistant to containment due to its nature as an abstract concept. If developers wish relief from technical debt, a better way must be found to describe the cost in real terms.
+
+There is a long list of potential consequences for allowing the unrestrained accrual of technical debt. A reduction in the ability to deliver features (usually referred to as 'velocity') is chief among them. Knowing the consequences of even the smallest change to the system is becomes difficult, with minor changes provide an opportunity to observe the butterfly effect. The engineer that coined the term technical debt, Ward Cunningham (*), explains that the core of technical debt is understanding. This lack of understanding has prevents engineers from safely modifying a system, and handicaps our ability to prevent outages. When those outages eventually occur (and they will), the knowledge deficit will frustrate any attempts at triage.
```

Comment:
This lack of understanding ~has~ prevents engineers from safely modifying a system